app/uk/gov/hmrc/agentsubscriptionfrontend/auth/AuthActions.scala

```diff
@@ -61,6 +62,26 @@ trait AuthActions extends AuthorisedFunctions with AuthRedirects with Monitoring
   def env = appConfig.environment
   def config = appConfig.configuration
 
+  def withSubscribedAgent[A](body: Arn => Future[Result])(
+    implicit request: Request[A],
+    hc: HeaderCarrier,
+    ec: ExecutionContext): Future[Result] =
+    authorised(Enrolment("HMRC-AS-AGENT") and AuthProviders(GovernmentGateway))
+      .retrieve(authorisedEnrolments) {
+        case enrolments =>
+          body(
+            getEnrolmentValue(enrolments, "HMRC-AS-AGENT", "AgentReferenceNumber")
+              .map(Arn(_))
+              .getOrElse(throw new InsufficientEnrolments("could not find the HMRC-AS-AGENT enrolment to continue")))
+      }
+      .recover {
+        case _: UnsupportedAffinityGroup =>
```

Comment:
Not sure this is a case anymore. But I think `authorised(Enrolment("HMRC-AS-AGENT")...` will throw an InsufficientEnrolments exception. We may need to handle that.